mautic/mautic
# Add headers on SendGrid emails sent via Sendgrid API

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ yes ]
| Deprecations?                          | [ no ]
| BC breaks? (use the c.x branch)        | [ no ]
| Automated tests included?              | [ code is called on addMetadataToMail on SendGridApiMessageTest.php  ] 
| Issue(s) addressed                     | Fixes #10688



#### Description:

This fix missing header `list-unsubscribe` when using transport set `Sendgrid API`.

New behavior to be discussed if should be merged: I'm removing `list-unsubscribe` if the `Precedence` is `Bulk`

#### Steps to test this PR:


1. Load up [this PR](https://mautibox.com)
2. See steps in the #10688 



<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10690"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
